test/vagrant-service-manager/proxy_test.rb

```diff
@@ -0,0 +1,80 @@
+require_relative '../test_helper'
+require 'vagrant/util/downloader'
+require 'uri'
+
+module VagrantPlugins
+  module ServiceManager
+    describe 'Proxy Test' do
+      before do
+        @machine = fake_machine
+
+        # Set test path
+        @plugin_test_path = "#{@machine.env.data_dir}/service-manager/test"
+        ServiceManager.temp_dir = "#{@plugin_test_path}/temp"
+        ServiceManager.bin_dir = "#{@plugin_test_path}/bin"
+
+        # Allow insecure curl
+        ENV['CURL_INSECURE'] = 'true'
+
+        # Set up the handler under test
+        options =  { type: :openshift, '--cli-version' => '1.2.0' }
+        @handler = ADBOpenshiftBinaryHandler.new(@machine, @machine.env, options)
+        @handler.build_download_url
+        @handler.validate_url
+        @handler.build_archive_path
+        @handler.ensure_binary_and_temp_directories
+      end
+
+      after do
+        FileUtils.rmtree(@plugin_test_path) if File.directory? @plugin_test_path
+        FakeHTTPSServer.stop
+        FakeHTTPProxyServer.stop
+      end
+
+      describe 'Unauthenticated proxy environment' do
+        before do
+          @http_proxy = FakeHTTPProxyServer.start ServiceManager.temp_dir
+          @https_server = FakeHTTPSServer.start
+
+          @machine.config.servicemanager.proxy = "http://localhost:#{FakeHTTPProxyServer::DEFAULT_PORT}"
+        end
+
+        it 'client binary download request should pass through proxy' do
+          @http_proxy.expect_http_connect('github.com:443')
+          dirname = File.dirname(__FILE__)
+          filename = File.join(dirname, '..', 'test_data', URI(@handler.url).path.split('/').last.to_s)
+          @https_server.expect_and_respond(@handler.url, filename)
```

Comment:
So, we are kind of making the stage here by filling `@url_request_received` by calling this line, right?